streaming/src/main/scala/org/apache/spark/streaming/receiver/Receiver.scala

```diff
@@ -256,6 +256,9 @@ abstract class Receiver[T](val storageLevel: StorageLevel) extends Serializable
   /** Identifier of the stream this receiver is associated with. */
   private var id: Int = -1
 
+  /** Location of the receiver for scheduling purposes. */
+  private[streaming] var host: Option[String] = None
```

Comment:
I am not sure why we really need a new API on Receiver just to attach a preferred location at the time of launching the receivers as tasks.